### PR TITLE
feat: show alarms in the NF9 UI

### PR DESCRIPTION
**Pull request by tjvantoll**
# Problem Context

The NF9 firmware creates `alarm.qo` notes when devices have a flow rate that is outside of the user’s configured flow thresholds. This PR adds code that parses the alarm.qo files, display them to the user in the UI, and allows users to clear all alarms.

## Screenshot (if applicable)

Screenshot 2022-12-08 at 4 09 46 PM

## Testing

I find it’s easiest to test by setting your device’s thresholds to nonsense values, which should generate alarms pretty quickly. Make sure you’re testing with the latest firmware so that you have a `reason` in your events.

## Ticket(s)

https://trello.com/c/eNMvrGTd/108-nf9-implement-alarms-in-ui

### CODE REVIEW

File: valve-monitor/web-app/src/pages/api/clear-alarms.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    case "POST":
+      try {
+        await services().getAppService().clearAlarms();
+      } catch (cause) {
+        throw new ErrorWithCause("Could not clear alarms", {
+          cause,
+        });
+      }
+
+      // res.status(StatusCodes.OK).json({});
```
Comment (paigen11) [inline valve-monitor/web-app/src/pages/api/clear-alarms.ts:21]: remove commented out code

File: valve-monitor/web-app/src/pages/index.tsx [hunk 2]
```diff
@@ -124,6 +135,17 @@ const Home: NextPage<HomeData> = ({ valveMonitorConfig, err }) => {
...
                         </h3>
+                        {valveMonitorDevices.filter(
+                          (device) => device.deviceAlarm
+                        ).length > 0 && (
+                          <Button
+                            type="primary"
+                            danger
+                            onClick={clearAllAlarms}
+                          >
+                            Clear Alarms
```
Comment (paigen11) [inline valve-monitor/web-app/src/pages/index.tsx:146]: disable this button after first click, I accidentally clicked it twice and while it cleared the alarms it also popped the error message at the top of the page

**Review by paigen11 — APPROVED** [review]
This looks really good.

I have two additional suggestions:

1. Give the error message some padding around the top of it so it doesn't run into the header.

Screen Shot 2022-12-09 at 9 31 54 AM

2. Add a sort function to the `valveMonitorDevices` variable in `index.tsx` so the table's always sorted alphabetically by device name when it renders
```
  const valveMonitorDevices = valveMonitorDeviceList?.sort((a, b) =>
    a.name.localeCompare(b.name)
  );
```

File: valve-monitor/web-app/src/pages/api/clear-alarms.ts (hunk 1 above)
Comment (tjvantoll) [reply valve-monitor/web-app/src/pages/api/clear-alarms.ts:21]: Ah, I accidentally committed my code to test the error scenario 😬 Good catch.

File: valve-monitor/web-app/src/pages/index.tsx (hunk 2 above)
Comment (tjvantoll) [reply valve-monitor/web-app/src/pages/index.tsx:146]: Good call. I added `disabled={isLoading}` which isn’t perfect but seems to work good enough for this.

**Comment by tjvantoll** [comment]
@paigen11 Good catches. I addressed the CSS, and added both your sorting code and Prisma fix. Merging now 🙂

**Pull request merged**